### PR TITLE
ProjectMilestones, ProjectTiers, and Terms (+ js unit tests)

### PR DESCRIPTION
**Pull request by JordoBrooks**

### CODE REVIEW

File: smart-contracts/ethereum/contracts/FundingService.sol [hunk 1]
```diff
@@ -92,6 +92,65 @@ contract FundingService {
...
+
+        verifyProjectMilestones(project);
+
+        verifyProjectTiers(project);
+
+        // Set project status to "Pending"
+        project.setStatus(Project.Statuses.Pending);
+    }
+
+    function verifyProjectMilestones(Project _project) private view {
```
[the remark this answers is not in the document]
Comment (JordoBrooks) [reply smart-contracts/ethereum/contracts/FundingService.sol]: @ericmuyser This should be covered in [this test](https://github.com/hyperbridge/funding-protocol/blob/9a29279c213a017012d5c797d65559b573e1ddbe/smart-contracts/ethereum/test/TestFundingService.js#L104)

File: smart-contracts/ethereum/contracts/FundingService.sol [hunk 2]
```diff
@@ -108,26 +108,14 @@ contract FundingService {
...
-        }
+    function verifyProjectMilestones(address _project) private view {
+        Project project = Project(_project);
 
         // If project has a timeline, verify:
         // - Milestones are present
         // - Milestone percentages add up to 100
-        if (!hasNoTimeline) {
-            uint timelineLength = _project.getTimelineMilestoneLength();
+        if (project.termIsActive(uint(Project.Term.NoTimeline))) {
```
Comment (ericmuyser) [inline smart-contracts/ethereum/contracts/FundingService.sol]: If we have to wrap in a uint prob better to define as uints rather than using enum

**Comment by ericmuyser** [comment]
I think the only other feedback from @0xwalid was we can store the terms as uint's, direct compare and ditch the loop

File: smart-contracts/ethereum/contracts/Project.sol [hunk 3]
```diff
@@ -5,15 +5,25 @@ contract Project {
...
+    struct ProjectTier {
+        uint contributorLimit;
+        uint minContribution;
+        uint maxContribution;
+        string rewards;
+    }
+
     enum Statuses {Draft, Pending, Published, Removed, Rejected}
 
+    enum Terms {NoRefunds, NoTimeline}
```
Comment (0xwalid) [inline smart-contracts/ethereum/contracts/Project.sol]: I think we should just use boolean flags here. easier to check and use since we only have two

**Pull request merged**